kornia/color/colormap.py

```diff
@@ -1,126 +1,121 @@
+
+        Returns:
+            Tensor: A tensor representing the colormap.
+        """
+        tensor_colors = tensor(list(base_colormap), dtype=self._dtype, device=self._device).T
+        return interpolate(tensor_colors[None, ...], size=num_colors, mode="linear")[0, ...]
 
     def __len__(self) -> int:
-        return self.colors.shape[-1]
+        r"""Returns the number of colors in the colormap.
 
+        Returns:
+            int: Number of colors in the colormap.
+        """
+        return self.colors.shape[-1]
 
-# TODO: Add more colormaps
-_BASE_AUTUMN: List[RGBColor] = [
-    [1.0, 0.0, 0.0],
-    [1.0, 0.01587301587301587, 0.0],
-    [1.0, 0.03174603174603174, 0.0],
-    [1.0, 0.04761904761904762, 0.0],
-    [1.0, 0.06349206349206349, 0.0],
-    [1.0, 0.07936507936507936, 0.0],
-    [1.0, 0.09523809523809523, 0.0],
-    [1.0, 0.1111111111111111, 0.0],
-    [1.0, 0.126984126984127, 0.0],
-    [1.0, 0.1428571428571428, 0.0],
-    [1.0, 0.1587301587301587, 0.0],
-    [1.0, 0.1746031746031746, 0.0],
-    [1.0, 0.1904761904761905, 0.0],
-    [1.0, 0.2063492063492063, 0.0],
-    [1.0, 0.2222222222222222, 0.0],
-    [1.0, 0.2380952380952381, 0.0],
-    [1.0, 0.253968253968254, 0.0],
-    [1.0, 0.2698412698412698, 0.0],
-    [1.0, 0.2857142857142857, 0.0],
-    [1.0, 0.3015873015873016, 0.0],
-    [1.0, 0.3174603174603174, 0.0],
-    [1.0, 0.3333333333333333, 0.0],
-    [1.0, 0.3492063492063492, 0.0],
-    [1.0, 0.3650793650793651, 0.0],
-    [1.0, 0.3809523809523809, 0.0],
-    [1.0, 0.3968253968253968, 0.0],
-    [1.0, 0.4126984126984127, 0.0],
-    [1.0, 0.4285714285714285, 0.0],
-    [1.0, 0.4444444444444444, 0.0],
-    [1.0, 0.4603174603174603, 0.0],
-    [1.0, 0.4761904761904762, 0.0],
-    [1.0, 0.492063492063492, 0.0],
-    [1.0, 0.5079365079365079, 0.0],
-    [1.0, 0.5238095238095238, 0.0],
-    [1.0, 0.5396825396825397, 0.0],
-    [1.0, 0.5555555555555556, 0.0],
-    [1.0, 0.5714285714285714, 0.0],
-    [1.0, 0.5873015873015873, 0.0],
-    [1.0, 0.6031746031746031, 0.0],
-    [1.0, 0.6190476190476191, 0.0],
-    [1.0, 0.6349206349206349, 0.0],
-    [1.0, 0.6507936507936508, 0.0],
-    [1.0, 0.6666666666666666, 0.0],
-    [1.0, 0.6825396825396826, 0.0],
-    [1.0, 0.6984126984126984, 0.0],
-    [1.0, 0.7142857142857143, 0.0],
-    [1.0, 0.7301587301587301, 0.0],
-    [1.0, 0.746031746031746, 0.0],
-    [1.0, 0.7619047619047619, 0.0],
-    [1.0, 0.7777777777777778, 0.0],
-    [1.0, 0.7936507936507936, 0.0],
-    [1.0, 0.8095238095238095, 0.0],
-    [1.0, 0.8253968253968254, 0.0],
-    [1.0, 0.8412698412698413, 0.0],
-    [1.0, 0.8571428571428571, 0.0],
-    [1.0, 0.873015873015873, 0.0],
-    [1.0, 0.8888888888888888, 0.0],
-    [1.0, 0.9047619047619048, 0.0],
-    [1.0, 0.9206349206349206, 0.0],
-    [1.0, 0.9365079365079365, 0.0],
-    [1.0, 0.9523809523809523, 0.0],
-    [1.0, 0.9682539682539683, 0.0],
-    [1.0, 0.9841269841269841, 0.0],
-    [1.0, 1.0, 0.0],
-]
-
-
-# Generate complete color map
-class AUTUMN(ColorMap):
```

Comment:
can you somehow make it still works, and add a deprecated warning about it?

Reply:
Ideally we shouldn't remove a functionality without minimal warning for kornia users

Reply:
https://github.com/kornia/kornia/blob/c8753f51c742cf1441b43fca94df1ce9a95a2cba/kornia/utils/helpers.py#L83

Reply:
Like a wrapper?
class AUTUMN(ColorMap) -> inside calling ColorMap('autumn')

Reply:
exactly, just keeping it working as it was in the last version and internally it can just call the new functions/module within the deprecated warning